src/coord/src/catalog.rs

```diff
@@ -406,101 +399,6 @@ impl CatalogState {
-// paths.
-#[derive(Debug, Clone)]
-struct CatalogEntryMap {
-    by_id: BTreeMap<GlobalId, CatalogEntry>,
-    /// Handle allowing operations on multiple persisted streams, either for
-    /// atomicity (writing a user transaction over multiple tables) or
-    /// efficiency (advancing the timestamp of all user tables in bulk).
-    ///
-    /// Invariant: This contains the same ids and handles as the set of tables
-    /// currently being persisted (i.e it matches the output of
-    /// self.generate_persist_multi_details() and is updated whenever the set of
-    /// persisted tables changes).
-    persist_multi_details: Option<TablePersistMultiDetails>,
-}
-
-impl std::ops::Deref for CatalogEntryMap {
-    type Target = BTreeMap<GlobalId, CatalogEntry>;
-    fn deref(&self) -> &Self::Target {
-        &self.by_id
-    }
-}
-
-impl CatalogEntryMap {
-    pub fn new(by_id: BTreeMap<GlobalId, CatalogEntry>) -> Self {
-        let mut ret = CatalogEntryMap {
-            by_id,
-            persist_multi_details: None,
-        };
-        ret.persist_multi_details = ret.generate_persist_multi_details();
-        ret
-    }
-
-    pub fn get_mut(&mut self, key: &GlobalId) -> Option<&mut CatalogEntry> {
-        // NB: Unfortunately, there's no way for this to regenerate
-        // self.persist_multi_details after the mutation is finished. However,
-        // as of when this was written, none of the callsites used this to
-        // change whether the entry is persisted, so we should be fine.
-        self.by_id.get_mut(key)
-    }
-
-    pub fn insert(&mut self, key: GlobalId, value: CatalogEntry) -> Option<CatalogEntry> {
-        let ret = self.by_id.insert(key, value);
-        self.persist_multi_details = self.generate_persist_multi_details();
-        ret
-    }
-
-    pub fn remove(&mut self, key: &GlobalId) -> Option<CatalogEntry> {
-        let ret = self.by_id.remove(key);
-        self.persist_multi_details = self.generate_persist_multi_details();
-        ret
-    }
-
-    fn generate_persist_multi_details(&self) -> Option<TablePersistMultiDetails> {
-        let mut all_table_ids = Vec::new();
-        let mut handles = Vec::new();
-        for (_, entry) in self.by_id.iter() {
-            match entry.item() {
-                CatalogItem::Table(Table {
-                    persist: Some(persist),
-                    ..
-                }) => {
-                    all_table_ids.push(persist.stream_id);
-                    handles.push(&persist.write_handle);
-                }
-                _ => {}
-            }
-        }
-        MultiWriteHandle::new_from_streams(handles.into_iter())
-            .ok()
-            .map(|write_handle| TablePersistMultiDetails {
-                all_table_ids,
-                write_handle,
-            })
-    }
-
-    pub fn persist_multi_details(&self) -> Option<&TablePersistMultiDetails> {
-        // Verify the persist_multi_details invariant.
-        debug_assert_eq!(
```

Comment:
I don't think we have an analog to this anymore. At least @danhhz thought this was important because he pointed it out to me on a recent PR.

Reply:
especially while only a subset of tables are being persisted, I think this is still a useful thing to check in debug. In the new world, I think it looks something like "set of tables in catalog with a persist name" == "set of tables in PersisterWithConfig::table_details". that should ensure we keep the necessary add_table and remove_table calls in all the right places

if it's too onerous to reintroduce this, though, no biggie

Reply:
No, easy enough! I added a soft assertion to `advance_local_inputs`, which seemed like an appropriately protective place.